### PR TITLE
added Zupass User Registry

### PR DESCRIPTION
**Pull request by crisgarner**

### CODE REVIEW

File: contracts/contracts/userRegistry/ZupassUserRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    view
+    returns (bool)
+  {
+    return users[_user];
+  }
+
+  /**
+    * @dev Check if the semaphore Id is verified.
+    */
+  function isVerifiedSemaphoreId(uint256 _semaphoreId)
```
Comment (yuetloo) [inline contracts/contracts/userRegistry/ZupassUserRegistry.sol:69]: do we need to check isVerifiedSemaphoreId() before adding the user in addUser()?

File: contracts/contracts/userRegistry/ZupassUserRegistry.sol [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    * @dev Add verified unique user to the registry.
+    */
+  function addUser(address _user, uint256 _semaphoreId)
+    external
+    onlyOwner
+  {
+    require(_user != address(0), 'UserRegistry: User address is zero');
+    require(!users[_user], 'UserRegistry: User already verified');
+    require(!semaphoreIds[_semaphoreId], 'UserRegistry: Semaphore Id already registred' );
+    users[_user] = true;
```
Comment (yuetloo) [inline contracts/contracts/userRegistry/ZupassUserRegistry.sol:33]: typo: `registred` should be `registered`

**Comment by yuetloo** [comment]
@crisgarner thank you for the PR.

File: contracts/contracts/userRegistry/ZupassUserRegistry.sol [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+pragma solidity ^0.6.12;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+
+import './IUserRegistry.sol';
+
+/**
+ * @dev A simple user registry managed by a trusted entity.
+ */
+contract ZupassUserRegistry is Ownable, IUserRegistry {
```
Comment (samajammin) [inline contracts/contracts/userRegistry/ZupassUserRegistry.sol:12]: Nitpick but I'd say `SemaphoreUserRegistry` would be more fitting for what this registry does

Comment (yuetloo) [reply contracts/contracts/userRegistry/ZupassUserRegistry.sol:12]: > Nitpick but I'd say `SemaphoreUserRegistry` would be more fitting for what this registry does

I'm happy to change the name to `SemaphoreUserRegistry`. This just means that the registry can be used to integrate with any system that can generate a unique semaphore id, not just by Zupass.

**Pull request merged**